library/core/src/iter/traits/iterator.rs

```diff
@@ -1495,6 +1495,14 @@ pub trait Iterator {
     /// assert_eq!(merged, "alphabetagamma");
     /// ```
     ///
+    /// Flattening also works on other types like Option and Result:
```

Comment:
It's not really specific to `Option` or `Result` though. It's that `Flatten` doesn't require `Item: Iterator`, it only needs  `Item: IntoIterator` and Option/Result are types that implement that.

Reply:
I think i do not really understand what you mean, should we not add this example should i change it?

Reply:
I'm pointing out that "types like Option and Result" is very vague. It doesn't tell the reader which those types have in common so they can be used with Flatten. I don't have a particular improvement in mind, I was just pointing out the vagueness and how it actually works.

Reply:
It seems like we already document that "iterator of iterators or an iterator of things that can be turned into iterators" above -- I'm not sure another example is all that useful. I think we have enough examples personally for flatten.

Reply:
I think this is okay, although its placement is interrupting the "you can also rewrite this" example below, so we should move this down. For wording, perhaps:

```suggestion
    /// Flattening works on any `IntoIterator` type, including `Option` and `Result`:
```

Although we're not *showing* `Result`, so maybe do so too, or remove that part.